### PR TITLE
Bootloader support

### PR DESCRIPTION
**Pull request by jacobmcnamee**
- [x] TODO: Update U-Boot repo and version in config

- Build uImage with compiled in rootfs and appended DTB
- Add script to generate image set for upgrades
- Reserve memory for remoteproc in DTS (makes mem=480M bootarg unnecessary)
- Update buildroot to 2016.08 (fixes issue where some kernel images were not copied to the output directory, see https://github.com/buildroot/buildroot/commit/b1f0212af66e374f2a5f88edd8d3f4b2b8a90d9b)

### CODE REVIEW

File: configs/piksiv3_defconfig [hunk 1]
```diff
@@ -40,19 +41,17 @@ BR2_PACKAGE_MTD_MKFSJFFS2=y
...
 BR2_TARGET_UBOOT_CUSTOM_GIT=y
 BR2_TARGET_UBOOT_CUSTOM_REPO_URL="https://github.com/jacobmcnamee/u-boot-xlnx.git"
-BR2_TARGET_UBOOT_CUSTOM_REPO_VERSION="swift-v2016.1"
-BR2_TARGET_UBOOT_BOARD_DEFCONFIG="zynq_microzed"
+BR2_TARGET_UBOOT_CUSTOM_REPO_VERSION="bootloader"
+BR2_TARGET_UBOOT_BOARD_DEFCONFIG="piksiv3_microzed_qspi"
 BR2_TARGET_UBOOT_NEEDS_DTC=y
 BR2_TARGET_UBOOT_FORMAT_IMG=y
-BR2_TARGET_UBOOT_SPL=y
-BR2_TARGET_UBOOT_SPL_NAME="spl/boot.bin"
```
Comment (jacobmcnamee) [inline configs/piksiv3_defconfig]: change this to `tpl/boot.bin` so it gets copied to the images folder

File: package/upgrade_tool/src/mtd.c [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+  }
+
+  /* seek to offset */
+  if (lseek(fd, offset, SEEK_SET) != offset) {
+    printf("error seeking flash\n");
+    return -1;
+  }
+
+  /* allocate read buffer */
+  void *buffer = malloc(BUFFER_SIZE);
```
Comment (gsmcmullin) [inline package/upgrade_tool/src/mtd.c]: This buffer could be on the stack

File: board/piksiv3/piksiv3_microzed.dts [hunk 3]
```diff
@@ -478,25 +478,37 @@
...
+    partition@qspi-img-std-a {
+      label = "qspi-img-std-a";
+      reg = <0x200000 0xbf0000>;
+    };
+    partition@qspi-img-std-b {
+      label = "qspi-img-std-b";
+      reg = <0xdf0000 0x10000>;
+    };
+    partition@qspi-img-failsafe {
+      label = "qspi-persistent";
```
Comment (gsmcmullin) [inline board/piksiv3/piksiv3_microzed.dts]: This should have a different label. `qspi-failsafe`?

Comment (jacobmcnamee) [inline board/piksiv3/piksiv3_microzed.dts]: Yeah, typo. `qspi-img-failsafe`

Comment (jacobmcnamee) [inline board/piksiv3/piksiv3_microzed.dts]: fixed now

File: package/upgrade_tool/src/mtd.c (hunk 2 above)
Comment (jacobmcnamee) [inline package/upgrade_tool/src/mtd.c]: Could, but it's already 64k and could potentially be larger. Do you see any issues with malloc? Or what would you have done?

**Pull request merged**